CONTRIBUTING.md

```diff
@@ -115,28 +144,231 @@ our project's repositories. The [Developer Certificate of Origin
+ suggestion. If the reviewer is asking a question, then usually the best way to answer
+ it is by improving the code or documentation. Answering it only in the code review
+ will not help future programmers after the PR is merged.
+
+Respond to the feedback by making changes in your working copy, committing them, and
+ periodically pushing them to GitHub when the tests pass locally. As soon as you
+ receive feedback, you can start working on it. The reviewer should assign the code
+ review back to you, but they might forget, so don't wait for that.
+
+After you have made the changes (in new commits please!), leave a comment asking the
+ the reviewer to take another look. If 3 business days go by with no review, it is okay
+ to bump. With the exception of rebasing discussed below, please do **not** force a push
+ with `git push -f` as it might cause loss of code review comments and context associated
+ with previous commits.
+
+When the PR is ready for merging, you may need to rebase your PR on top of the current codebase.
+This can be done via the PR's GitHub Web page or with the CLI as either `git pull --rebase`
+or `git fetch` and `git rebase` from `upstream/main`.
+If there are any conflicts please fix them locally before committing and pushing to the PR branch.
+If you're unsure on how to resolve a conflict, please ask the reviewer for guidance or help.
+This `git push` is the only instance where you are expected to use `-f` or `--force`. Note that
+it might "litter" your commit with unrelated commits done on `main` since the branch was created
+and thus make reviewing more difficult. We therefore strive to [make PR small and reviews quick](#how-to-get-your-prs-reviewed-quickly),
+to avoid too much divergence and merge conflict.
+
+<!--
+The repository owner can prevent incorrect pull request merges. In the repository settings, in the “Merge button” section, disable “Allow merge commits” and “Allow rebase merging”. You might also want to enable “Automatically delete head branches”.
+-->
+
+After you have addressed all the review feedback, explicitly request a re-review.
+ Do not assume that person will know when you are done. There are many ways to request
+ a re-review:
+
+* Add the reviewer to the PR on Github - this works even if that person has reviewed the
+ pull request before.
+* Assign the pull request to that person, using the “Assignees” list.
+* Write a comment in the conversation in the GitHub pull request.
+
+1. When a pull request has been approved, the reviewer will squash and merge your
+ commits. If you prefer to rebase your own commits, at any time leave a comment on the
+ pull request to let them know that.
+
+1. At this point your changes are available to be included in the next release of
+ ClusterLink! After your first pull request is merged, you will be invited to the
+ Contributors team (TODO create Github team) which you may choose to accept (or not).
+ Joining the team lets you have issues in GitHub assigned to you.
```

Comment:
I would remove this.

Reply:
This is meant as a way to recognize contributor efforts. It does not grant maintainer rights.
AFAIK, you can't assign issues (or mark them as assigned to yourself) if you're not somehow part of the repo's contributors list

Reply:
I don't think you want to ALWAYS want to add anyone who upstreamed a commit.
Perhaps they're not even interested in that honor.
And on the other direction, the team may not want to necessarily add anyone who upstreamed a change.
This also relates to my "nit: trivial" comments.
I don't think it makes a good impression if you say "we accept anyone who makes a single commit".

Reply:
Some projects do that, others do not. Some even extend an offer of maintainership to committers. Submitters are free to refuse...
I'm not proposing maintainers, only contributor. They can also decide to join the developer mailing list...

Assuming you want to acknowledge contributions publicly, you would typically do this in
- contributors file, listing everyone
- listing relevant contributions in release notes
- adding them to contributors mailing list.

We can change it to >1 commit, but if someone wants to help out with the project on a more ongoing basis, you may want to assign issues to them at some point.